lib/Agrammon/OutputFormatter/GUI.pm6

```diff
@@ -67,6 +56,38 @@ sub get-data($model, $outputs) {
     return @records;
 }
 
+sub make-record($fq-name, $output, $model, $raw-value, $var, $filters?) {
+    my $format = $model.output-format($fq-name, $output);
+    my $full-value = flat-value($raw-value);
+    my $value = ($format  && $full-value.defined) ?? sprintf($format, $full-value)
+                                                  !! $full-value;
+    return %(
+        :$format,
+        :print($model.output-print($fq-name, $output)),
+        :order($model.output-order($fq-name, $output)),
+        :labels($model.output-labels($fq-name, $output)),
+        :units($model.output-units($fq-name, $output)),
+        :fullValue($full-value),
+        :$value,
+        :$var,
+        :$filters,
+    );
+}
+
+sub push-filters(@records, $fq-name, $output, $model,
+                 Agrammon::Outputs::FilterGroupCollection $collection,
+                 $var) {
+    for $collection.results-by-filter-group {
+        my %filters := .key;
+        my $value := .value;
+        my @filters = %filters.map: { .key ~ '=' ~ .value };
+        for @filters.kv -> $idx, $filter-id {
+            push @records, make-record($fq-name, $output, $model, $value, $var, %filters);
+        }
```

Comment:
I don't think this makes sense. We're emitting `%filters` into the record (good, as it means we're sending them as structured data). But if it's a complex filter (multiple inputs contribute to the filter) we'll emit the same `%filters` once for each input. We're also doing throwaway work that we don't emit. I think this code should be:

```
    for $collection.results-by-filter-group {
        my %filters := .key;
        my $value := .value;
        push @records, make-record($fq-name, $output, $model, $value, $var, %filters);
    }
```

Reply:
Fixed, I am not sure what I was thinking when doing this, if anything ...
![Screenshot 2020-10-05 at 21 30 01](https://user-images.githubusercontent.com/198038/95123893-f2f83e80-0752-11eb-8056-e7825528aaea.png)